### ISSUE TITLE
Network for registration?

### ISSUE THREAD

**Issue by pvtodorov** [broadinstitute/Tangram#2]
Hi, 

In the manuscript there is mention and demonstration (Fig 6, 7, 8) of a NN used to determine the coronal depth and automatically register slices to the ABA CCF. Is the automatic image registration and automated region calling capability available as part of Tangram?

Thanks!

**Comment by lewlin** [comment]
Tangram is, strictly speaking, the algorithm for mapping, but the NN will be available soon (ie tomorrow at the latest).

Keep an eye out on our team repos: https://github.com/orgs/broadinstitute/teams/tommaso_team/

We will make public a new repo soon.

**Comment by pvtodorov** [comment]
Just wanted to check if the repo was ever updated - the provided link comes up as a 404

Totally understand if there has been a delay!

**Comment by lewlin** [comment]
Hi pvtodorov,

I am sorry the URL with the team repos yields a 404: it's for Broad Institute only and I did not realize that.
Yes, we uploaded the code for the NN shortly after your request: you can find the code here:

https://github.com/broadinstitute/one-shot-atlas

In this repo you'll find the code of our Siamese although you'll notice that we have not released our training set yet. We will do that after the paper is published. 

To make sure we are on the same page: our NN fetches the mouse brain image at the corresponding depth. This is our first step our pipeline. The second is to generate masks using a U-net model (we use this one https://github.com/qubvel/segmentation_models). Again, we will release the training set with the labels soon (probably after the paper is published). To register the masks, we use ANTs (see here http://stnava.github.io/ANTs/).

Therefore, our NN does not do end-to-end registration. If you are looking for a DL model for doing that, I recommend you check out voxelmorph (https://github.com/voxelmorph/voxelmorph).

Ciao,
